spec/forms/totp_verification_form_spec.rb

```diff
@@ -33,5 +33,23 @@
         expect(form.submit).to eq result
       end
     end
+
+    context 'when the code is not exactly Devise.otp_length digits' do
+      it 'returns FormResponse with success: false' do
+        user = build_stubbed(:user)
+        codes = %W(123abc 1234567 abcdef aaaaa\n123456\naaaaaaaaa)
```

Comment:
I see what you mean. I was trying to make this dynamic based on the Devise setting, in case it changes, but the wrong codes themselves are not derived from the setting, so I think in this case it would be clearer to just say 6 digits, and if the setting changes, we'll update the spec. What do you think?